src/Antiforgery/test/AntiforgeryMiddlewareTest.cs

```diff
@@ -40,6 +44,8 @@ public async Task RespectsIgnoreAntiforgeryMetadata()
     [InlineData("TRACE")]
     [InlineData("HEAD")]
     [InlineData("OPTIONS")]
+    [InlineData("DELETE")]
```

Comment:
I thought DELETE needed CSRF because it was a write, so I did a quick $search_engine and came across this [SO answer](https://stackoverflow.com/a/11972282/1064169).

It seems to suggest that you don't need it for PUT or DELETE, but then there's some hand-waving about it maybe being prudent to cover it anyway.

Based on that, should it either cover PUT and DELETE or cover neither?

Or is there a better answer that explains why PUT yes but DELETE no for my own interest?

Reply:
CSRF only protects you from a third-party site sending a form, which it can only do with GET/POST, so I think we are good.

Reply:
This is the check that MVC does:
```
        if (HttpMethods.IsGet(method) ||
            HttpMethods.IsHead(method) ||
            HttpMethods.IsTrace(method) ||
            HttpMethods.IsOptions(method))
```